sdk/python/kfp/v2/google/aiplatform.py

```diff
@@ -0,0 +1,464 @@
+      input_name].artifact_type.CopyFrom(
+        type_utils.get_artifact_type_schema_message(art.param_type))
+
+  for output_name, param_type in output_parameters.items():
+    pipeline_component_spec.output_definitions.parameters[
+      output_name].type = type_utils.get_parameter_type(param_type)
+
+  for output_name, artifact_type in output_artifacts.items():
+    pipeline_component_spec.output_definitions.artifacts[
+      output_name].artifact_type.CopyFrom(artifact_type.get_ir_type())
+
+  pipeline_component_spec.executor_label = dsl_utils.sanitize_executor_label(
+      task_name)
+
+  # Iterate through the inputs/outputs specs to get pipeline task spec.
+  for input_name, param in input_parameters.items():
+    if isinstance(param, dsl.PipelineParam) and param.op_name:
+      # If the param has a valid op_name, this should be a pipeline parameter
+      # produced by an upstream task.
+      pipeline_task_spec.inputs.parameters[input_name].CopyFrom(
+          pipeline_spec_pb2.TaskInputsSpec.InputParameterSpec(
+              task_output_parameter=pipeline_spec_pb2.TaskInputsSpec.InputParameterSpec.TaskOutputParameterSpec(
+                  producer_task='task-{}'.format(param.op_name),
+                  output_parameter_key=param.name
+              )))
+    elif isinstance(param, dsl.PipelineParam) and not param.op_name:
+      # If a valid op_name is missing, this should be a pipeline parameter.
+      pipeline_task_spec.inputs.parameters[input_name].CopyFrom(
+          pipeline_spec_pb2.TaskInputsSpec.InputParameterSpec(
+              component_input_parameter=param.name))
+    else:
+      # If this is not a pipeline param, then it should be a value.
+      pipeline_task_spec.inputs.parameters[input_name].CopyFrom(
+          pipeline_spec_pb2.TaskInputsSpec.InputParameterSpec(
+              runtime_value=pipeline_spec_pb2.ValueOrRuntimeParameter(
+                  constant_value=dsl_utils.get_value(param))))
+
+  for input_name, art in input_artifacts.items():
+    if art.op_name:
+      # If the param has a valid op_name, this should be an artifact produced
+      # by an upstream task.
+      pipeline_task_spec.inputs.artifacts[input_name].CopyFrom(
+          pipeline_spec_pb2.TaskInputsSpec.InputArtifactSpec(
+              task_output_artifact=pipeline_spec_pb2.TaskInputsSpec.InputArtifactSpec.TaskOutputArtifactSpec(
+                  producer_task='task-{}'.format(art.op_name),
+                  output_artifact_key=art.name)))
+    else:
+      # Otherwise, this should be from the input of the subdag.
+      pipeline_task_spec.inputs.artifacts[input_name].CopyFrom(
+          pipeline_spec_pb2.TaskInputsSpec.InputArtifactSpec(
+              component_input_artifact=art.name
+          ))
+
+  # TODO: Add task dependencies/trigger policies/caching/iterator
+  pipeline_task_spec.component_ref.name = dsl_utils.sanitize_component_name(
+      task_name)
+
+  # Construct dummy I/O declaration for the op.
```

Comment:
This is because `ContainerOp` class has a I/O declaration ties to container implementation. It does not provide a way to declare I/O and then wire the I/O with the underlying container implementation.

That's why we need to construct the dummy I/O by filling in dummy local path and specify them in the `ContainerOp`.

Reply:
Ideally we should refactor out a new `BaseOp` which is implementation-neutral, but I don't think we have time for that in short term.

Reply:
Thanks for explaining.